aip/general/0202.md

```diff
@@ -0,0 +1,103 @@
+---
+id: 202
+state: draft
+created: 2023-09-14
+placement:
+  category: fields
+  order: 0
+---
+
+# Fields
+
+The [`google.api.Field`][field proto] type, through its accompanying extension
+`google.api.field`, enriches a field's schema beyond the basic name and type
+information.
+
+## Guidance
+
+Decorating a field with `google.api.field` is only necessary when explicitly
+stated in this AIP or another that leverages `google.api.Field` information. As
+such, the guidance herein applies to those scenarios as well.
+
+### Format
+
+Fields with a primitive type can still have a specific format. To convey that
+type format, the `Field.Format` enumeration is used via the
+`(google.api.field).format` extension field. The following guidance conveys the
+meaning of and requirements for use of each `Field.Format` value.
+
+#### UUID4
+
+The `UUID4` format represents a UUID version 4 value as governed by
+[RFC 4122][]. It **must** only be used on a field of type `string`.
+
+Such a value **may** be normalized by the service to entirely lowercase letters.
+For example, the value `F47AC10B-58CC-0372-8567-0E02B2C3D479` would be
+normalized to `f47ac10b-58cc-0372-8567-0e02b2c3d479`.
+
+#### IPv4
+
+The `IPV4` format represents an IP v4 address as governed by [RFC 791][]. It
+**must** only be used on a field of type `string`.
+
+Such a value **may** be condensed by the service, with leading zeros in each
+octet stripped. For example, `001.022.233.040` would be condensed to
+`1.22.233.40`.
+
+#### IPv6
+
+The `IPV6` format represents an IP v6 address as governed by [RFC 2460][]. It
+**must** only be used on a field of type `string`.
+
+Such a value **may** be normalized by the service to entirely lowercase letters,
+and zero-padded partial and empty octets. For example, the value `2001:DB8::`
+would be normalized to `2001:0db8:0000:0000`.
+
+#### IPv6 or IPv6
+
+The `IPV4_OR_IPV6` value indicates that the field can be either an IP v4 or v6
+address, as described in the [IPv4](#ipv4) and [IPv6](#ipv6) sections.
+
+#### Format Compatibility
+
+Adding a format specifier to an existing, unspecified field **is** backwards
+compatible.
```

Comment:
... is backwards compatible if the content of the field is already conforming to the format's specification.

e.g. it's back-compat if it's describing a fact that is already true. It's breaking if the effect is to take a free-form unique string and limit it to UUID values.

Reply:
Yeah @amanda-tarafa raised the same point and I figure we need to include that exception, it's just impossible to lint/static check for.

Reply:
Done